### ISSUE TITLE
LLVM's maxnum builder is broken in LLVM6

### ISSUE THREAD

**Issue by gnzlbg** [rust-lang/rust#49261]
This was fixed in this LLVM commit: https://github.com/llvm-mirror/llvm/commit/e271f8c46c6bee271f75848b6110588da5df7d2f

Once we upgrade to LLVM 7 a couple of fail tests are going to start passing because of this. 

This issue is referenced in all parts of the code that need updating.

**Comment by cuviper** [comment]
When testing with LLVM master, I just hit a test "failure" on `run-fail/simd-intrinsic-float-minmax.rs`, because the run actually passed!  IMO this is the wrong use of `run-fail` -- these should be tests that are *supposed* to fail, not just "known to fail because of a bug."

Wouldn't it be better to make that test a `run-pass`, marked with `min-llvm-version 7.0`?  Or maybe `run-pass` with the known-broken parts FIXME-commented out?

**Comment by gnzlbg** [comment]
@cuviper that makes sense. 

> Wouldn't it be better to make that test a run-pass, marked with min-llvm-version 7.0?

Now that we know the test passes with master, then yes. But I did not test it with master, so I did not know. I wanted that to be loud the first time somebody did that so that we can move the test to run-pass. I did not thought about somebody testing with master though, I thought this would just happen when we tried to upgrade to LLVM7.

Now that we know the fix in master really did fix this, moving it to run-pass with min-llvm-version 7.0 makes sense.
